src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -216,8 +217,10 @@ public List<String> collectAllDatabasePaths() {
     }
 
     public void addSearchHistory(String search) {
-        searchHistory.remove(search);
-        searchHistory.add(search);
+        if (!GlobalSearchBar.searchBoxFocused.get()) {
```

Comment:
I would not add this here. The state manager should not have a dependency on the search bar. This is also not relevant here.

Reply:
At first I just attached the change listener inside GlobalSearchBar, but I was concerned about the case we call the addSearchHistory from elsewhere, and passed the bound object to StateManager.

So, is there any chance the method to be called from another class to record a query?

Reply:
No, why should we? Tip: Every IDE has a thing called "Call Hierarchy" to show where the method is called. Use this to check where the method is called

Reply:
I placed the change listener under constructor. I'm not sure what to do inside the if statement:
```
searchBoxFocused.bind(searchField.focusedProperty());
searchBoxFocused.addListener((observable, oldPropertyValue, newPropertyValue) -> {
    if (!newPropertyValue) {
        // do sth
    }
});
```
About performSearch method:
```
if (!(searchBoxFocused.get() || searchField.textProperty().get().isEmpty())) {
    this.stateManager.addSearchHistory(searchField.textProperty().get());
}

Reply:
The focus listener gives you two values:
old and new
aka: the previous value and the new value:
So: When the focus is lost: check if old = true && new = false  
=> The user has left the search box
=> Now we call the addSearchHistory with the value of the     this.stateManager.addSearchHistory(searchField.textProperty().get());

Reply:
Yes I get it. 

The change listener is placed under constructor. The method addSearchHistory, in contrast, is called inside performSearch(), so that's the reason I use binding to attach the value of the focus property to another BooleanProperty object's value.

As you can see from the two blocks of code (my previous comment), both if statements do the same job. Checking whether the focus is lost or not. So that's why I'm wondering what to do in the first if statement.

Reply:
In this case you neither need the binding nor the listener.

My approach would have been
```
searchBoxFocused.addListener((obs, oldValue, newValue) -> {
    if (oldValue  && !new  &&  && ! searchField.textProperty().get().isEmpty())) {
    this.stateManager.addSearchHistory(searchField.textProperty().get());
    }
});
```
remove the existing one in peformSearch

Reply:
> In this case you neither need the binding nor the listener.

Nice, it seems to works correctly. I will write some javadoc (regarding
the classes I touched) and unit tests.

> remove the existing one in performSearch

Actually, removing `addSearchHistory` method call from the`performSearch` 
prevents the unnecessary  recording of history every the time the `searchTask` timer
exceeds the  (`SEARCH_DELAY = 400`)  and `performSearch` is called.